### ISSUE TITLE
Mail Server

### ISSUE THREAD

**Issue by saeedvaziry** [vitodeploy/vito#302]
From: https://vitodeploy.featurebase.app/p/mail-server

Enable VitoDeploy to install and provision a mail server from scratch

**Comment by xDefcon** [comment]
https://stalw.art/ may be a good option to evaluate. It’s the “new tech” for mail servers. Downside is that It does not come with a webmail panel such as roundcube.

**Comment by SiFuMax09** [comment]
But Roundcube can be installed separately.

**Comment by dyanakiev** [comment]
Creating and managing a custom mail server can be quite a hassle. I’ve been exploring alternatives, and I think Docker support could be a game-changer. We could offer a predefined list of services (via docker-compose files) for example, Mailcow as a ready-to-deploy mail server stack and also let users create and manage their own setups, similar to how Portainer handles containers.

**Comment by J4Wx** [comment]
@dyanakiev 

> Creating and managing a custom mail server can be quite a hassle. I’ve been exploring alternatives, and I think Docker support could be a game-changer. We could offer a predefined list of services (via docker-compose files) for example, Mailcow as a ready-to-deploy mail server stack and also let users create and manage their own setups, similar to how Portainer handles containers.
> 
> One downside to note is that Docker doesn’t play well with UFW by default, as it tends to bypass firewall rules and expose all published ports, which would require additional configuration for proper security.

I agree with this. Custom mail servers are a nuisance. But being able to add a docker "service" to a server, and then add docker-compose files (especially if the user could provide custom ones at their own risk) would not only solve this problem, but provide a vast array of possibilities moving forwards. 

It would certainly be an "advanced" feature (not something that the average user would use at all), but still of significant value imo.

**Comment by saeedvaziry** [comment]
Docker will be in the next major release 🤞 (No ETA/roadmap yet)

**Comment by saeedvaziry** [comment]
Closing as there will be no docker implemented.
